policy.tf

```diff
@@ -0,0 +1,371 @@
+data "aws_iam_policy_document" "s3_bucket_with_kms_policy_document_1" {
+  policy_id = "${var.bucket_iam_user}Policy1"
+
+  statement {
+    sid    = "IAMS3BucketPermissions"
+    effect = "Allow"
+
+    resources = [
+      "${aws_s3_bucket.s3_bucket.arn}",
+    ]
+
+    actions = [
+      "s3:ListBucket",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "s3_bucket_with_kms_policy_document_2" {
```

Comment:
A policy document can contain multiple statements. By separating these out you're duplicating the amount of code doing count checks and attaching iam policies.

Reply:
Think this is a good start and I'll refactor the code later. Just don't want to add more complexity on to an already complex change.